caikit_nlp/resources/pretrained_model/hf_auto_causal_lm.py

```diff
@@ -168,6 +169,18 @@ def _get_data_collator(self, **kwargs) -> "transformers.DataCollator":
                 Collator to be used for causal language modeling.
         """
 
+        if "train_on_completion" in kwargs and kwargs["train_on_completion"]:
+            applicable_args = ["mlm", "response_template", "instruction_template"]
+            collator_kwargs = {
+                key: kwargs[key] for key in applicable_args if key in kwargs
+            }
+
+            if "mlm" not in collator_kwargs:
+                collator_kwargs["mlm"] = False
+
+            return DataCollatorForCompletionOnlyLM(
+                tokenizer=self._tokenizer, return_tensors="pt", **collator_kwargs
```

Comment:
Have you tested this with multiGPU by any chance? In the past, we've seen some kind of strange behavior for some of the collators around dynamic padding, curious if that's been observed here

Reply:
Raghu's team has tested it with MultiGPU, but the plan is to only use this codebase for single GPU going forward. MultiGPu will go through non caikit path